src/classes/STG_PanelRDHealthCheck_TEST.cls

```diff
@@ -0,0 +1,285 @@
+                npe03__Record_Type__c = UTIL_RecordTypes_TEST.getDefaultRecordTypeId(Opportunity.SObjectType)
+            )
+        );
+
+        STG_PanelHealthCheck_CTRL ctrl = new STG_PanelHealthCheck_CTRL();
+
+        TDTM_Config_API.disableAllTriggers();
+
+        STG_PanelRDHealthCheck.verifyRecurringDonationConfiguration(ctrl);
+        for (STG_PanelHealthCheck_CTRL.DetectResult dr : ctrl.listDR) {
+            System.assertEquals(STG_PanelHealthCheck_CTRL.statusSuccess, dr.strStatus,
+                'The result status should have been "' + STG_PanelHealthCheck_CTRL.statusSuccess + '".\n' + dr
+            );
+        }
+    }
+
+    /**
+     * @description Validate that the legacy health check fails with an error when a custom installment period
+     * is created, but does not have a matching picklist value in the installment period picklist
+     */
+    @IsTest
+    private static void shouldValidateLegacyWithCustomInstallmentPeriods() {
+        npe03__Custom_Installment_Settings__c customInstallment = new npe03__Custom_Installment_Settings__c(
+            npe03__Increment__c = 'Weeks',
+            npe03__Value__c = 2,
+            Name = BAD_PICKLIST_VALUE
+        );
+        insert customInstallment;
+
+        npe03__Recurring_Donations_Settings__c rds = UTIL_CustomSettingsFacade.getRecurringDonationsSettingsForTest(
+            new npe03__Recurring_Donations_Settings__c(
+                IsRecurringDonations2Enabled__c = false,
+                npe03__Opportunity_Forecast_Months__c = 1,
+                npe03__Maximum_Donations__c = 20,
+                npe03__Open_Opportunity_Behavior__c = RD2_Constants.CloseActions.Mark_Opportunities_Closed_Lost.name(),
+                npe03__Record_Type__c = UTIL_RecordTypes_TEST.getDefaultRecordTypeId(Opportunity.SObjectType)
+            )
+        );
+
+        STG_PanelHealthCheck_CTRL ctrl = new STG_PanelHealthCheck_CTRL();
+
+        TDTM_Config_API.disableAllTriggers();
+
+        Test.startTest();
+
+        STG_PanelRDHealthCheck.verifyRecurringDonationConfiguration(ctrl);
+
+        String results = getValidationResults(ctrl);
+        System.assert(results.contains(BAD_PICKLIST_VALUE),
+            'The error should reference the "' + BAD_PICKLIST_VALUE + '" picklist value.\n' + results
+        );
+
+        Test.stopTest();
+    }
+
+    /**
+     * @description Validates that Enhanced Recurring Donations is configured properly in the org.
+     * Because RD1 is the default for NPSP, there are some settings that will fail the health check
+     * such as for the Quarterly installment period. Situations like this are validated for in the
+     * method.
+     */
+    @IsTest
+    private static void shouldValidateEnhancedRecurringDonationsConfig() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        npe03__Recurring_Donations_Settings__c rds = UTIL_CustomSettingsFacade.getRecurringDonationsSettingsForTest(
```

Comment:
Settings creation can be reused as per [RD2_Settings_TEST.setUpConfiguration()](https://github.com/SalesforceFoundation/NPSP/blob/master/src/classes/RD2_Settings_TEST.cls#L191) while extra fields (if any) can be set in the input Map.